homeassistant/components/onkyo/config_flow.py

```diff
@@ -0,0 +1,281 @@
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the configuration of a single receiver."""
+        _LOGGER.debug("Configure receiver info: %s", self._receiver_info)
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self._receiver_info.model_name,
+                data={
+                    CONF_HOST: self._receiver_info.host,
+                    CONF_VOLUME_RESOLUTION: user_input[CONF_VOLUME_RESOLUTION],
+                },
+                options={
+                    OPTION_MAX_VOLUME: OPTION_MAX_VOLUME_DEFAULT,
+                    # Preseed the sources with a dictionary where keys and values are equal.
+                    OPTION_SOURCES: dict(
+                        zip(
+                            user_input[OPTION_SOURCES],
+                            user_input[OPTION_SOURCES],
+                            strict=False,
+                        )
+                    ),
+                },
+            )
+
+        unique_id = self._receiver_info.identifier
+        await self.async_set_unique_id(unique_id, raise_on_progress=False)
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self._receiver_info.host}
+        )
+
+        return self.async_show_form(
+            step_id="configure_receiver",
+            description_placeholders={
+                "name": f"{self._receiver_info.model_name} ({self._receiver_info.host})"
+            },
+            data_schema=STEP_CONFIGURE_SCHEMA,
+        )
+
+    async def async_step_import(self, user_input: dict[str, Any]) -> ConfigFlowResult:
+        """Import the yaml config."""
+        _LOGGER.debug("Import flow user input: %s", user_input)
+
+        host: str = user_input[CONF_HOST]
+        name: str = user_input[CONF_NAME]
+        max_volume: float = float(user_input[OPTION_MAX_VOLUME])
+        volume_resolution: int = user_input[CONF_RECEIVER_MAX_VOLUME]
+        sources: dict[str, str] = user_input[OPTION_SOURCES]
+
+        info: ReceiverInfo | None
+        try:
+            info = await async_interview(host)
+        except Exception:
+            _LOGGER.exception("Import flow interview error for host %s", host)
+            return self.async_abort(reason="cannot_connect")
+
+        if info is None:
+            _LOGGER.error("Import flow interview error for host %s", host)
+            return self.async_abort(reason="cannot_connect")
+
+        unique_id = info.identifier
+        await self.async_set_unique_id(unique_id, raise_on_progress=False)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=name,
+            data={
+                CONF_HOST: host,
+                CONF_VOLUME_RESOLUTION: volume_resolution,
+            },
+            options={
+                OPTION_MAX_VOLUME: max_volume,
+                OPTION_SOURCES: sources,
+            },
+        )
```

Comment:
These values need to be validated first, and, if needed, transformed to a proper form.

Reply:
Fixed myself. Can be marked resolved.